src/Configuration/Types.hs

```diff
@@ -0,0 +1,383 @@
+    arbitrary = do
+        osName              <- elements ["windows64", "macos64", "linux64"]
+        osConfigurationYaml <- genSafeText
+        osInstallDirectory  <- genSafeText
+        osX509ToolPath      <- genSafeText
+        osNodeArgs          <- arbitrary
+        osPass              <- arbitrary
+
+        pure OSConfig{..}
+
+-- | Node arguments
+data NodeArgs = NodeArgs
+    { naKeyfile          :: !Text
+    , naLogsPrefix       :: !Text
+    , naTopology         :: !Text
+    , naUpdateLatestPath :: !Text
+    , naWalletDBPath     :: !Text
+    , naTlsPath          :: !Text
+    } deriving (Eq, Generic, Show)
+
+instance Interpret NodeArgs where
+    autoWith opt = interpretAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 2}
+
+instance Inject NodeArgs where
+    injectWith opt = injectAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 2}
+
+instance Arbitrary NodeArgs where
+    arbitrary = do
+        naKeyfile          <- genSafeText
+        naLogsPrefix       <- genSafeText
+        naTopology         <- genSafeText
+        naUpdateLatestPath <- genSafeText
+        naWalletDBPath     <- genSafeText
+        naTlsPath          <- genSafeText
+
+        pure NodeArgs{..}
+
+-- | Paths
+data Pass = Pass
+    { pStatePath           :: !Text
+    , pNodePath            :: !Text
+    , pNodeDbPath          :: !Text
+    , pNodeLogConfig       :: !Text
+    , pNodeLogPath         :: !(Maybe Text)
+    , pWalletPath          :: !Text
+    , pWalletLogging       :: !Bool
+    , pWorkingDir          :: !Text
+    , pFrontendOnlyMode    :: !Bool
+    , pUpdaterPath         :: !Text
+    , pUpdaterArgs         :: ![Text]
+    , pUpdateArchive       :: !(Maybe Text)
+    , pUpdateWindowsRunner :: !(Maybe Text)
+    , pLauncherLogsPrefix  :: !Text
+    } deriving (Eq, Generic, Show)
+
+instance Interpret Pass where
+    autoWith opt = interpretAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 1}
+
+instance Inject Pass where
+    injectWith opt = injectAutoWithOption $ opt {fieldModifier = lowerHead . T.drop 1}
+
+instance Arbitrary Pass where
+    arbitrary = do
+        pStatePath           <- genSafeText
+        pNodePath            <- genSafeText
+        pNodeDbPath          <- genSafeText
+        pNodeLogConfig       <- genSafeText
+        pNodeLogPath         <- maybeOf genSafeText
+        pWalletPath          <- genSafeText
+        pWalletLogging       <- arbitrary
+        pWorkingDir          <- genSafeText
+        pFrontendOnlyMode    <- arbitrary
+        pUpdaterPath         <- genSafeText
+        pUpdaterArgs         <- listOf genSafeText
+        pUpdateArchive       <- maybeOf genSafeText
+        pUpdateWindowsRunner <- maybeOf genSafeText
+        pLauncherLogsPrefix  <- genSafeText
+
+        pure Pass{..}
+-- | Launcher configuration
+data LauncherConfig = LauncherConfig
+    { lcfgFilePath    :: !Text
+    , lcfgKey         :: !Text
+    , lcfgSystemStart :: !(Maybe Natural)
```

Comment:
Oh? Is there no way to convert the `Natural` to a plain `Integer` here? I understand you had issues with the types, but still looks kind of funny - not really sure if the author intended to tie our hands using only `Natural` and not something idiomatic like `Integer`.

Reply:
Well, there's two ways to do this:
1. Modify the content of dhall files so it can be parsed as `Integer`. So I'll need to modify something like this:

```
-- Add plus
walletPort = +8090
````

2. Bump the dhall's version to the latest one.
The current version of dhall that the shell use (1.5.1) does not have much freedom when it comes to defining an instance of `Inject` which is a instance needed to encode given data type into dhall format.
The latest version of dhall (1.9.1) on the other hand seems to have this feature (http://hackage.haskell.org/package/dhall-1.19.1/docs/Dhall.html#v:-62--42--60-).
To do so, I'd need to either 
1. Add an lot of `extra-deps` into the shell project
2. Use nix package system to use certain version of `dhall` package

Reply:
For now, I'll apply step 1 since it's the most safest and easiest one.

Reply:
Sounds like the most straightforward solution.